opentracing-api/src/main/java/io/opentracing/ActiveSpanManager.java

```diff
@@ -0,0 +1,34 @@
+package io.opentracing;
+
+/**
+ * XXX(bhs): comment
+ */
+public interface ActiveSpanManager {
+    // XXX(bhs): good comment. just a marker, obviously.
+    public interface Snapshot {
+        Span span();
+    }
+
+    /**
+     * Returns the span associated with the current execution context.
+     *
+     */
+    Span active();
+
+    /**
+     * Retrieve the associated SpanSnapshot.
+     * @return the SpanSnapshot that encapsulates Span state that should propagate across in-process concurrency boundaries.
+     */
+    Snapshot snapshot(Span span);
+
+    /**
+     * Activates a given Snapshot (per snapshot()).
+     */
+    Span activate(Snapshot snapshot);
+
+    /**
+     * XXX: comment
+     */
+    void deactivate(Snapshot snapshot);
```

Comment:
I don't love that this requires a Snapshot (rather than just a Span), but doing a Span has bad implications for the impl.